### DISCUSSION TITLE
Proxy Support

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#11387] [Feature Requests]
In our institute of my university we want to use directus behind a proxy server. Unfortunatly there is no possibility to configure proxy settings.

Here is hopefully a solution, but it is not very user friendly...
https://github.com/directus/directus/discussions/6386

**Comment by aidenfoxx** [comment]
Global-agent should do the trick? https://www.npmjs.com/package/global-agent

**Comment by NilsBaumgartner1994** [comment]
I am already using it. I fear that Directus (grant) requires https because of oAuth 2.0 right ?


Beste Grüße,
Nils Baumgartner

**Reply by aidenfoxx** [reply]
Directus uses `openid-client` and it should work correctly with `global-agent`.

**Reply by rijkvanzanten** [reply]
Directus no longer relies on grant. The requirement on https depends on whatever third party you're trying to integrate with.

**Reply by NilsBaumgartner1994** [reply]
I am using global-agent

```
export GLOBAL_AGENT_HTTP_PROXY=http://www-proxy.rz.uni-osnabrueck.de:80 && 
export GLOBAL_AGENT_HTTPS_PROXY=http://www-proxy.rz.uni-osnabrueck.de:80 && 
npx directus bootstrap && 
npx directus start
```

but the problem still stays:

```
hiwisdirectus | 16:42:25 ✨ request completed GET 302 /auth/login/studiptest?redirect=http://hiwis.informatik.uni-osnabrueck.de/hiwis/api/admin/login?continue 11ms
hiwisdirectus | 16:42:47 ⚠️  Invalid user credentials.
hiwisdirectus | 16:42:47 ✨ request completed GET 302 /auth/login/studiptest/callback?code=b8119570e7c8a5ec96e98674b711c9df2fde6fb1&state=XXXXXXXXGiZd6Sc4C4LX4_9M0-PstNn8gYDzDK4 3.5s
```

**Reply by aidenfoxx** [reply]
You need to initialize "global-agent". Try something like this:

```
export GLOBAL_AGENT_HTTP_PROXY=http://www-proxy.rz.uni-osnabrueck.de:80 && 
export GLOBAL_AGENT_HTTPS_PROXY=http://www-proxy.rz.uni-osnabrueck.de:80 &&
npx directus bootstrap && 
node -r 'global-agent/bootstrap' node_modules/directus/dist/start.js
```

**Reply by NilsBaumgartner1994** [reply]
Thanks ! That was the point !

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
